server/handlers/post.js

```diff
@@ -38,20 +38,16 @@ export const getPostList = (req, res, next) => {
  * @type {RouteHandler}
  */
 export const getPost = async (req, res) => {
-  let postData = {};
   await findPostById(req.params.postId).then(returnPost => {
     if (returnPost.post) {
       return res.status(404).json({ error: returnPost.post });
     }
-    postData.post = returnPost;
-    postData.comment = [];
+    const postData = { ...returnPost._doc };
+    postData.commentList = [];
     findCommentByPostId(returnPost._id)
       .then(post => {
-        if (post.comment) {
-          return res.status(404).json({ error: post.comment });
-        }
         post.forEach(doc => {
-          postData.comment.push(doc);
+          postData.commentList.push(doc);
```

Comment:
I'm confused what we're mapping here... is this just converting an object to an array? maybe add some types here so it's clear what's happening. I think theres stuff here that could be converted to a 1-liner from lodash like
postData.commentList = values(post)

Reply:
The data being returned from the findCommentByPostId is an array. Taking that into account, I might be able to just spread it's contents into postData.commentList.

Reply:
fixed